### PR TITLE
Sophos | Use `Timezone` config when invalid IANA timezone in events

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
- WHAT: Uses `Timezone` parameter from the config to parse the time values when invalid IANA `timezone` is present in the log.
- WHY:  In the current scenario, a sample log message such as `date=2023-10-27 time=12:11:45 timezone="EDT" device_name="XG330"...` would parse into a `@timestamp` value of `"2023-10-27T12:11:45.000Z"` which is wrong. This happens because `EDT` is not present in IANA timezone list. This change will utilize `Timezone` parameter from the config when the `timezone` from the log is invalid. 

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

**Comment by efd6** [comment]
Suggest something like this without the new config surface.
```diff
diff --git a/packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml b/packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml
index 8532e2c19..5469b3026 100644
--- a/packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml
+++ b/packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml
@@ -74,12 +74,6 @@ processors:
 - set:
     field: event.timezone
     copy_from: _temp_.tz
-# Even at this stage, user can override the timezone with `tz_override` config option.
-- set:
-    field: event.timezone
-    copy_from: _conf.tz_offset
-    override: true
-    if: ctx._conf?.tz_offset != null && ctx._conf.tz_offset != 'local' && ctx._conf.tz_override != null && ctx._conf.tz_override == true
 - gsub:
     field: event.timezone
     pattern: '([+-][0-9]{2})([0-9]{2})'
@@ -109,13 +103,48 @@ processors:
       - ISO8601
     if: ctx._temp_?.time != null
     on_failure:
+      # Try to re-parse as configured timezone or UTC if unknown.
       - remove:
           field: event.timezone
           ignore_missing: true
-      # Try to re-parse as UTC to catch when TZ is invalid or unknown.
+      - set:
+          field: event.timezone
+          copy_from: _conf.tz_offset
+          ignore_empty_value: true
+      - date:
+          tag: "date_config_tz_fallback"
+          timezone: event.timezone
+          if: ctx.event?.timezone != null
+          field: _temp_.time
+          formats:
+            - yyyy-MM-dd HH:mm:ss
+            - yyyy-MM-dd HH:mm:ss Z
+            - yyyy-MM-dd HH:mm:ss z
+            - ISO8601
+          on_failure:
+            - remove:
+                field: event.timezone
+                ignore_missing: true
+            # Try to re-parse as UTC to catch when TZ is invalid or unknown.
+            - date:
+                tag: "date_utc_config_tz_fallback"
+                timezone: UTC
+                field: _temp_.time
+                formats:
+                  - yyyy-MM-dd HH:mm:ss
+                  - yyyy-MM-dd HH:mm:ss Z
+                  - yyyy-MM-dd HH:mm:ss z
+                  - ISO8601
+                on_failure:
+                  - append:
+                      field: error.message
+                      value: "fail-{{{ _ingest.on_failure_processor_tag }}}"
+                  - fail:
+                      message: "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message: {{ _ingest.on_failure_message }}"
       - date:
           tag: "date_utc_fallback"
           timezone: UTC
+          if: ctx.event?.timezone == null
           field: _temp_.time
           formats:
             - yyyy-MM-dd HH:mm:ss
```

**Comment by kcreddy** [comment]
That does make sense. Thanks Dan!

**Review by P1llus — APPROVED** [review]
A slightly different approach than what I had in mind, but this should work as well 👍 Thanks for adding it in

**Comment by kcreddy** [comment]
@efd6 could you give your approval if the change now looks good?

File: packages/sophos/data_stream/xg/_dev/test/pipeline/test-common-config.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ fields:
     - preserve_original_event
   _conf:
     default: defaulttest.local
+    # tz_offset: "-0400"
```
Comment (efd6) [inline packages/sophos/data_stream/xg/_dev/test/pipeline/test-common-config.yml:6]: Is this intended?

File: packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -103,13 +103,48 @@ processors:
       - ISO8601
     if: ctx._temp_?.time != null
     on_failure:
+    # Try to re-parse as configured timezone or UTC if unknown.
```
Comment (efd6) [inline packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml:106]:
```suggestion
      # Try to re-parse as configured timezone or UTC if unknown.
```

**Review by efd6 — APPROVED** [review]
nits then LGTM

File: packages/sophos/data_stream/xg/_dev/test/pipeline/test-common-config.yml (hunk 1 above)
Comment (kcreddy) [reply packages/sophos/data_stream/xg/_dev/test/pipeline/test-common-config.yml]: no. my bad again

File: packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (kcreddy) [reply packages/sophos/data_stream/xg/elasticsearch/ingest_pipeline/default.yml]: updated

**Pull request merged**